frontend/src/components/mainPage/codeEditor/subcomponents/Codearea.vue

```diff
@@ -42,6 +43,72 @@ export default {
             this.$emit("pickVariableEvent", variable);
         },
 
+        moveTrackedVariables(changes) {
+            const handleVarTrackerMove = (ch, varObj) => {
+                if (ch.size > 0) {
+                    if (ch.start > varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                } else {
+                    if (ch.start >= varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.end > varObj.start && ch.end < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                }
+            };
+
```

Comment:
W przypadku tego "zdarzenia" "Delete" nie usuwałbym całego sceneObject, tylko usunąłbym pole variable z tego obiektu. Chodzi o przypadek, że użytkownik zdefiniował sobie cały obiekt grafowy z kilkoma właściwościami poustawiał sobie kolorki, a po przypadkowym usunięciu zmiennej z kodu cały obiekt mu znika. Jak tylko usuniemy powiązanie obiektu ze zmienną to najwyżej sobie ją powiąże na nowo. Zadanie "Walidacja obiektu scenicznego" można by rozszerzyć o dodatkową walidację przed uruchomieniem programu i przed zapisem całego projektu. Ale to ta walidacja już chyba po stronie backendu musiała być.